### PR TITLE
it.mijorus.webappmanager

### PR DESCRIPTION
**Pull request by mijorus**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
https://github.com/linuxmint/webapp-manager/issues/227

Waiting for response...

- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Answers from:
https://github.com/flathub/flathub/pull/3823#event-8253692249

- Why are these needed?
```
        "--filesystem=home",
        "--talk-name=org.freedesktop.Flatpak"
```
Yes, the app tests the existence of some system files and writes desktop files in the `./local/share` folder

- The App ID is also wrong. If you're packaging a Linux Mint Program, you need to use com.linuxmint, not your own name

I would like to fork this project and add new features, not just packaging it (if Flathub's rules allows it...). The linux mint team does not seem interested in releasing it as a Flatpak

Let me know, thank you

### CODE REVIEW

**Comment by barthalion** [comment]
If you consider it a full fork, you don't need Mint's permission.

**Comment by barthalion** [comment]
>  writes desktop files in the ./local/share folder

There's a portal for that. Please see how it's implemented in Epiphany source code.

> the app tests the existence of some system files 

Which system files?

**Comment by mijorus** [comment]
> > writes desktop files in the ./local/share folder
> 
> There's a portal for that. Please see how it's implemented in Epiphany source code

I'm looking for that, thank you. I've been lazy there.

> > the app tests the existence of some system files
> 
> Which system files?

This app needs to know if certain programs are installed (browsers) and needs to know their executable path.
To do this, it  runs the `test` command against a list of pre-loaded paths (like `/usr/bin/firefox`)

That's it.

I don't know any other way to to this inside a flatpak, because the `/usr/` directory is always overwritten by Flatpak's sandbox, no matter what permissions I give

**Comment by JakobDev** [comment]
> I don't know any other way to to this inside a flatpak, because the /usr/ directory is always overwritten by Flatpak's sandbox, no matter what permissions I give

If you gave the right permission, you can access the host usr under /run/host/usr inside of the Sandbox

**Comment by mijorus** [comment]
Permissions have been updated; about the `host:ro`, this is the best I managed to do

**Comment by hfiguiere** [comment]
bot, build it.mijorus.webappmanager

File: it.mijorus.webappmanager.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        "python3-requirements.json",
```
Comment (hfiguiere) [inline it.mijorus.webappmanager.json:29]: file is missing.

**Comment by mijorus** [comment]
bot, build it.mijorus.webappmanager

**Comment by mijorus** [comment]
bot, build it.mijorus.webappmanager

**Comment by barthalion** [comment]
Trying to add a web app ends with this on my end:

```
Traceback (most recent call last):
  File "/app/share/webapp-manager/webapp_manager/WebappManagerWindow.py", line 279, in on_ok_button
    self.manager.create_webapp(name, url, icon, category, browser, custom_parameters, isolate_profile, navbar, privatewindow)
  File "/app/share/webapp-manager/webapp_manager/common.py", line 262, in create_webapp
    shutil.copy(icon, new_icon)
  File "/usr/lib/python3.10/shutil.py", line 417, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.10/shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'webapp-manager'
```

It makes an entry appear on the list when I close the dialog window, but the desktop launcher also appears to be wrong:

```
Exec=epiphany --application-mode  --profile="/var/home/bpiotrowski/.local/share/org.gnome.Epiphany.WebApp-WP0798" "https://wp.pl"
```

I don't have Epiphany installed and all my browsers are installed via Flatpak.

**Comment by mijorus** [comment]
> I don't have Epiphany installed and all my browsers are installed via Flatpak.

I have identified a possible fix for the "File not found" issue and working on a fix, btw the fact that you were able to create a webapp for a browser you don't have it's another bug as well.

Are you sure that you do not have epiphany installed? It comes bundled with many distros under different names

**Comment by mijorus** [comment]
bot, build it.mijorus.webappmanager

**Comment by mijorus** [comment]
hi, @barthalion could you please test this new build?

**Comment by mijorus** [comment]
hi @barthalion any updates on this one?

**Comment by barthalion** [comment]
bot, build it.mijorus.webappmanager

**Comment by barthalion** [comment]
It still behaves exactly the same on my end.

**Comment by mijorus** [comment]
> It still behaves exactly the same on my end.

I'm testing with a clean VM on both ubuntu and fedora.
I found some issues regarding epiphany and the isolated profile mode, which I'm working on.

I will update this PR when the issues are fixed.

Please let me know if you  find anything else.

Thanks for your time

**Pull request closed without merging**